<launch>

    <arg name="namespace" default="" />

    <group ns="$(arg namespace)">
        <node pkg="mapper"
            type="mapper"
            name="mapper_node"
            output="screen">

            <!-- Namespace -->
            <param name="namespace" value="$(arg namespace)"/>

            <!-- Set of depth camera topics -->
            <param name="depth_cam_prefix" value="/"/>
            <rosparam param="depth_cam_names"> ["openni_camera_link"] </rosparam>
            <param name="depth_cam_suffix" value="/depth/points"/>

            <!-- Inertial frame name (used for getting tf tree information) -->
            <param name="inertial_frame_id" value="/map"/>

            <!-- Set parameters for the mapper -->
            <param name="map_resolution" value="0.15"/>          <!-- meters -->
            <param name="max_range" value="10"/>                 <!-- meters -->
            <param name="min_range" value="1.00"/>              <!-- meters -->

            <!-- Time to forget information (set as negative if infinity) -->
            <param name="memory_time" value="-30.0"/>           <!-- seconds -->

            <!-- Fading memory thread update rate (Hz) -->
            <param name="fading_memory_update_rate" value="1"/><!-- Hz -->

            <!-- Map inflation parameters -->
            <param name="inflate_radius" value="0.25"/>       <!-- meters -->

            <!-- Camera FOV (radians): map is only updated within this FOV -->
            <param name="cam_fov" value="0.90"/>             <!-- radians -->

            <!-- Camera's width divided by height (unitless) -->
            <param name="cam_aspect_ratio" value="1.3099"/>

            <!-- Map probabilities -->
            <param name="occupancy_threshold" value="0.5"/>     <!-- 0-1 -->
            <param name="probability_hit" value="0.7"/>         <!-- 0-1 -->
            <param name="probability_miss" value="0.25"/>       <!-- 0-1 -->
            <param name="clamping_threshold_min" value="0.1"/>  <!-- 0-1 -->
            <param name="clamping_threshold_max" value="0.9"/>  <!-- 0-1 -->

            <!-- Collision Checking parameters -->
            <param name="traj_compression_max_dev" value="0.01"/>     <!-- meters -->
            <param name="traj_compression_resolution" value="0.02"/>  <!-- meters -->
            <param name="collision_check_rate" value="0.01"/>         <!-- Hz -->

            <!-- Frequency at which tf listeners update tf -->
            <param name="tf_update_rate" value="50"/>            <!-- Hz -->

            <!-- Service names -->
            <param name="update_resolution" value="mapper/update_resolution"/>
            <param name="update_memory_time" value="mapper/update_memory_time"/>
            <param name="update_inflation_radius" value="mapper/update_inflation_radius"/>
            <param name="reset_map" value="mapper/reset_map"/>

            <!-- Publisher names -->
            <param name="obstacle_markers" value="mapper/obstacle_markers"/>
            <param name="free_space_markers" value="mapper/free_space_markers"/>
            <param name="inflated_obstacle_markers" value="mapper/inflated_obstacle_markers"/>
            <param name="inflated_free_space_markers" value="mapper/inflated_free_space_markers"/>
            <param name="frustum_markers" value="mapper/frustum_markers"/>
            <param name="discrete_trajectory_markers" value="mapper/discrete_trajectory_markers"/>
            <param name="collision_detection" value="mapper/collisions"/>

            <!-- Path of mapper in computer: used for saving/loading maps -->
            <param name="local_path" value="$(find mapper)"/>

        </node>

    </group>
</launch>